sequencer/src/l1_client.rs

```diff
@@ -129,6 +130,34 @@ impl L1Client {
             }
         }
     }
+    /// Get fee info for each `Deposit` ocurring between `prev`
+    /// and `new`. Returns `Vec<FeeInfo>`
+    pub async fn get_pending_deposits(
+        &self,
+        prev_finalized: Option<u64>,
+        new_finalized: u64,
+        address: Address,
+    ) -> Vec<FeeInfo> {
+        let prev = prev_finalized.unwrap_or(0);
+        // When called from `Header::new` during testing both will be
+        // `0`. BUG: https://github.com/foundry-rs/foundry/issues/7245
+        // We treat this as if there were no deposits.
+        if prev == new_finalized {
+            return vec![];
+        }
+
+        // query for deposit events, errors are handled as "no deposits".
```

Comment:
Since we can't construct a valid block header if we don't have the correct deposits I think we should retry on errors instead.

Reply:
fixed in d85c24bb94c26ad18d1fb202f110d67f6f90de3e